### PR TITLE
Add *_MATRIX_LED_COUNT generation

### PR DESCRIPTION
**Pull request by zvecr**
## Description

~~Current hitting preprocessor isues:~~
```
⚠ acheron/apollo/87h/gamma: RGB_MATRIX_LED_COUNT->rgb_matrix.led_count: invalid literal for int() with base 10: 'DRIVER_1_LED_TOTAL'
⚠ aeboards/satellite/rev1: RGB_MATRIX_LED_COUNT->rgb_matrix.led_count: invalid literal for int() with base 10: 'ISSI_DRIVER_TOTAL'
```

~~Parsing is currently disabled due to the above issue. A 2nd pass will refactor keyboards, but this at least allows dd configuration in the short term.~~

Parsing is no longer important as the value is generated only from the length of the LED array.

## Types of Changes

- [x] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by t-8ch** [comment]
Would it make sense to calculate the number from the layout if no explicit value is specified in `info.json`?

**Review by drashna — APPROVED** [review]

File: data/schemas/keyboard.jsonschema [hunk 1]
```diff
@@ -374,6 +374,7 @@
                     }
                 },
                 "driver": {"type": "string"},
+                "led_count": {"$ref": "qmk.definitions.v1#/unsigned_int"},
```
Comment (fauxpark) [inline data/schemas/keyboard.jsonschema:377]: Is there any situation in which these would be needed?

Comment (zvecr) [reply data/schemas/keyboard.jsonschema:377]: Maybe? The only case that comes to mind in the short term, when there is inconsistency in led count and layout data. Where existing keyboards potentially rely on the backing array being filled with zeros. But maybe all of these were refactored. No strong feelings against it's removal.

**Review by dunk2k — APPROVED** [review]

File: data/schemas/keyboard.jsonschema (hunk 1 above)
Comment (zvecr) [reply data/schemas/keyboard.jsonschema]: Removed logic

**Comment by Kriechi** [comment]
Should this also account for split configs?
https://github.com/qmk/qmk_firmware/blob/7a737235ffd49c32d2c5561e8fe53fd96baa7f96/data/schemas/keyboard.jsonschema#L438
and
https://github.com/qmk/qmk_firmware/blob/7a737235ffd49c32d2c5561e8fe53fd96baa7f96/data/schemas/keyboard.jsonschema#L503

If not defined by user, assume `split_count = [ len(...) / 2 , len(...) / 2 ] ` ?

**Comment by fauxpark** [comment]
Would need some additional logic to error out if `RGB_MATRIX_LED_COUNT` is odd.

**Comment by zvecr** [comment]
> Should this also account for split configs?

Not in scope for the changes in this PR.

**Pull request merged**